src/m3ninx/index/segment/builder/multi_segments_terms_iter.go

```diff
@@ -121,67 +121,68 @@ func (i *termsIterFromSegments) setField(field []byte) error {
 }
 
 func (i *termsIterFromSegments) Next() bool {
-	if i.err != nil {
-		return false
-	}
-
-	if !i.keyIter.Next() {
-		return false
-	}
+	for {
```

Comment:
Note to others: the diff becomes tiny after hiding whitespace changes ([`?w=1`](https://github.com/m3db/m3/pull/3807/files?w=1))

Reply:
wow did not know this github hack!